operator/cmd/mesh/manifest-validate.go

```diff
@@ -0,0 +1,345 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/operator/pkg/validate"
+	"istio.io/istio/pkg/config/validation"
+)
+
+const (
+	traceSamplingMin float64 = 0.0
+	traceSamplingMax float64 = 100.0
+)
+
+var (
+	// Boolean values for --set flags
+	boolValues = []bool{true, false}
+
+	// Ref: https://kubernetes.io/docs/concepts/configuration/overview/#container-images
+	imagePullPolicy = []string{"Always", "IfNotPresent", "Never"}
+
+	// Keep this list updated as per following
+	// https://preliminary.istio.io/docs/setup/additional-setup/config-profiles/
+	profile = []string{"default", "demo", "empty", "minimal", "preview", "remote", "separate"}
```

Comment:
I think the user can provide their own profiles when using --installPackagePath. @ostromart is this correct?

When run without this change I already see proper validation:
```
$ manifest apply -y --set profile=foobar
Error: failed to apply manifests: failed to generate manifest: unsupported Profile type: foobar
exit status 1
```

is there additional value we get from this?

Reply:
Nothing exceptional here, it just provides what profiles they can use if they pass wrong one. So they don't have to go and [search valid values for the profile](https://istio.io/docs/setup/additional-setup/config-profiles/).


```go
 $ istioctl manifest apply --set profile=foobar

Error: failed to apply manifests: 
 Unsupported value: "foobar", supported values for: "profile" is "default, demo, empty, minimal, preview, remote, separate"
```
But if user can provide their custom profiles, it needs to be updated to handle this case.

Reply:
This won't work properly because the allowed values depend on the scanned file list in operator/data/profiles.